src/main/scala/mesosphere/marathon/api/HttpBindings.scala

```diff
@@ -0,0 +1,48 @@
+package mesosphere.marathon
+package api
+
+import com.sun.jersey.spi.container.servlet.ServletContainer
+import java.util.EnumSet
+import javax.servlet.{ DispatcherType, Filter, Servlet }
+import org.eclipse.jetty.servlet.{ FilterHolder, ServletContextHandler, ServletHolder }
+import org.eclipse.jetty.servlets.EventSourceServlet
+
+/**
+  * Used to bind the Marathon root application (api routes, etc.), various filters and other servlets to the Jetty
+  * instance.
+  */
+object HttpBindings {
+  def apply(
+    handler: ServletContextHandler,
+    rootApplication: RootApplication,
+    leaderProxyFilter: LeaderProxyFilter,
+    limitConcurrentRequestsFilter: LimitConcurrentRequestsFilter,
+    corsFilter: CORSFilter,
+    cacheDisablingFilter: CacheDisablingFilter,
+    eventSourceServlet: EventSourceServlet,
+    webJarServlet: WebJarServlet,
+    publicServlet: PublicServlet): Unit = {
+
+    val allDispatchers = EnumSet.allOf(classOf[DispatcherType])
+    def addFilter(path: String, filter: Filter, dispatches: EnumSet[DispatcherType] = allDispatchers): Unit = {
+      handler.addFilter(new FilterHolder(filter), path, dispatches)
+    }
+
+    def addServlet(path: String, servlet: Servlet): Unit = {
+      handler.addServlet(new ServletHolder(servlet), path)
+    }
+
+    addFilter("/*", leaderProxyFilter)
+    addFilter("/*", limitConcurrentRequestsFilter)
+    addFilter("/*", corsFilter)
+    addFilter("/*", cacheDisablingFilter)
+    addServlet("/v2/events", eventSourceServlet)
+
+    for { p <- Seq("/ui", "/ui/*", "/help", "/api-console", "/api-console/*") } {
```

Comment:
First time I found out that we have an `api-console` endpoint and it's actually working! O_O